src/common/include/common/ec.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+#ifndef _COMMON_EC_H
+#define _COMMON_EC_H
+
+#ifndef __EC__
+#error __EC__ must be defined
+#endif
+
+#define it5570e 5570
```

Comment:
Is this required? The scratch ROM uses the correct SCAR registers depending on __EC__ definition without having these

Reply:
Yes, this is needed. The code in scratch.c actually doesn't work how you think.
As a quick example, I modified scratch.c as follows:
<code> 
#if \_\_EC__ == it5570e
</code>
<code>
+#error it5570e
</code>
and
<code>
 #elif \_\_EC__ == it8587e
</code>
<code>
+#error it8587e
</code>

When building galp3-c, the following error is reported:
<code>
sdcc -mmcs51 --model-large --code-size 65536 --Werror -Isrc/common/include -D__FIRMWARE_VERSION__=2020-08-02_2e6487d-dirty -Isrc/board/system76/galp3-c/include -D__BOARD__=system76/galp3-c -DI2C_SMBUS=I2C_0 -DPS2_TOUCHPAD=PS2_3 -DCHARGER_CHARGE_CURRENT=1536 -DCHARGER_CHARGE_VOLTAGE=13056 -DCHARGER_INPUT_CURRENT=1920 -DLEVEL=4 -Isrc/board/system76/common/include -DSCRATCH_OFFSET=1024 -DSCRATCH_SIZE=1024 -Ibuild/system76/galp3-c/2020-08-02_2e6487d-dirty/include -DFLASH_OFFSET=2048 -DFLASH_SIZE=1024 -Ibuild/system76/galp3-c/2020-08-02_2e6487d-dirty/include -Isrc/ec/it8587e/include -D__EC__=it8587e -Isrc/arch/8051/include -D__ARCH__=8051 -o build/system76/galp3-c/2020-08-02_2e6487d-dirty/board/system76/common/scratch.rel -c src/board/system76/common/scratch.c
src/board/system76/common/scratch.c:15:2: error: #error it5570e
</code>
You'll node that  #error it5570e is reported (we are in the 5570e case), even though \_\_EC__ is defined as 8587.

The reason this is happening is because the compiler is that both it8587e and it5570e are not defiend - that is, they have the same value. As a result, only the first case ever happens in scratch.c.

Reply:
Also note that adding the new include in causes the expected error to be printed:
#include <common/ec.h>
src/board/system76/common/scratch.c:25:2: error: #error it8587e